ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/services/admin/ApiPartyManagementService.scala

```diff
@@ -57,55 +65,53 @@ class ApiPartyManagementService private (
     *
     * Checks invariants and forwards the original result after the party is found to be persisted.
     *
-    * @param result The result of the party allocation
     * @return The result of the party allocation received originally, wrapped in a [[Future]]
     */
-  private def pollUntilPersisted(result: AllocatePartyResponse): Future[AllocatePartyResponse] = {
-    require(result.partyDetails.isDefined, "Party allocation response must have the party details")
-    val newParty = result.partyDetails.get.party
-    val description = s"party $newParty"
-
-    PollingUtils
-      .pollUntilPersisted(partyManagementService.listParties _)(
-        _.exists(_.party == newParty),
-        description,
-        50.milliseconds,
-        500.milliseconds,
-        d => d * 2,
-        scheduler,
-        loggerFactory)
-      .map { numberOfAttempts =>
-        logger.debug(s"Party $newParty available, read after $numberOfAttempts attempt(s)")
-        result
-      }(DE)
+  private def pollUntilPersisted(submissionId: SubmissionId): Future[PartyEntry] = {
+    partyManagementService
+      .partyEntries()
```

Comment:
@gerolf-da this is gonna poll through the entire set of parties looking for the matching submissionID, potentially 10s of thousands of entries in real world / ASX case.  we had previously changed the query to just get back by matching submissionId, i guess you went with this approach because it is closer to Source stream done for ledger entries?

Reply:
one option is to poll starting from the offset at submission time. I'm trying that out for the config admin service.

Reply:
Good point. I pushed a change that only streams the party entries from the ledger end before the party allocation request is made.